File: EcoChallengeBackend/EcoChallenge.Services/Database/Entities/Organization.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EcoChallenge.Services.Database.Entities
{
    public class Organization
    {
        [Key]
        public int Id { get; set; }
        [Required, Column("name"), MaxLength(100)]
        public string? Name { get; set; }
        [Column("description")]
        public string? Description { get; set; }
        [Column("website"), MaxLength(255)]
        public string? Website { get; set; }
        [Column("logo_url"), MaxLength(255)] 
        public string? LogoUrl { get; set; }

        [Column("contact_email"), MaxLength(100)]
        public string? ContactEmail { get; set; }
        [Column("contact_phone"), MaxLength(20)]
        public string? ContactPhone { get; set; }
        [Column("category"), MaxLength(50)]
        public string? Category { get; set; }
        [Column("is_verified")]
        public bool IsVerified { get; set; } = false;
        [Column("is_active")]
        public bool IsActive { get; set; } = true;
        [Column("created_at")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        [Column("updated_at")]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        // Navigation properties
        public virtual ICollection<Donation>? Donations { get; set; }
    }
}

File: EcoChallengeFrontend/ecochallenge_mobile/lib/models/organization.dart
class Organization {
  final int id;
  final String? name;
  final String? description;
  final String? website;
  final String? logoUrl;
  final String? contactEmail;
  final String? contactPhone;
  final String? category;
  final bool isVerified;
  final bool isActive;
  final DateTime createdAt;
  final DateTime updatedAt;

  Organization({
    required this.id,
    this.name,
    this.description,
    this.website,
    this.logoUrl,
    this.contactEmail,
    this.contactPhone,
    this.category,
    required this.isVerified,
    required this.isActive,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Organization.fromJson(Map<String, dynamic> json) {
    return Organization(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      website: json['website'],
      logoUrl: json['logoUrl'],
      contactEmail: json['contactEmail'],
      contactPhone: json['contactPhone'],
      category: json['category'],
      isVerified: json['isVerified'],
      isActive: json['isActive'],
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: DateTime.parse(json['updatedAt']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'website': website,
      'logoUrl': logoUrl,
      'contactEmail': contactEmail,
      'contactPhone': contactPhone,
      'category': category,
      'isVerified': isVerified,
      'isActive': isActive,
      'createdAt': createdAt.toIso8601String(),
      'updatedAt': updatedAt.toIso8601String(),
    };
  }
}
